mne/beamformer/_dics.py

```diff
@@ -421,6 +423,97 @@ def apply_dics_epochs(epochs, filters, return_generator=False, verbose=None):
     return stcs
 
 
+@verbose
+def apply_dics_tfr(epochs, filters, freqs, method='morlet', average=False,
+                   n_csd=256, n_overlap=0, verbose=None, **csd_kwargs):
+    """Apply Dynamic Imaging of Coherent Sources (DICS) beamformer weights.
+
+    Apply Dynamic Imaging of Coherent Sources (DICS) beamformer weights
+    on single trial time-frequency data.
+
+    Parameters
+    ----------
+    epochs : Epochs
+        Single trial epochs.
+    filters : instance of Beamformer
+        DICS spatial filter (beamformer weights)
+        Filter weights returned from :func:`make_dics`.
+    freqs : ndarray, shape (n_freqs,)
+        The frequencies over which to compute cross-spectral density.
+    method : 'morlet' | 'fourier' | 'multitaper'
+        The cross-spectral density method to use.
```

Comment:
Both frequencies and method should be informed by the filter, as they should match.

Reply:
Good catch!